cloud/shared/bin/lib/config_loader.py

```diff
@@ -214,50 +215,52 @@ def _validate_infra_variables(
 
           We are calling the GitHub API with unauthenticated requests, which are rate-limited.
           The rate limit allows for up to 60 requests per hour associated with the originating 
           IP address.
         """
-        tag = tag.strip()
-        if tag == 'latest':
-            # Translate "latest" into a version number
-            tag = self._get_latest_version_number()
 
-        release_url = f"https://api.github.com/repos/civiform/civiform/git/refs/tags/{tag}"
-        release_response = requests.get(release_url)
+        tag = tag.strip()
 
-        if release_response.status_code == 200:
-            return self._get_commit_sha_for_tag(
-                release_response.json()["object"]["sha"])
-        else:
-            raise self.VersionNotFoundError(
-                f"The commit sha for version {tag} could not be found. Are you using a valid tag such as latest or a valid version number like v1.0.0? {release_response.status_code} - {release_response.json()['message']}"
-            )
+        try:
+            if tag == "latest":
+                return self._fetch_json_val(
+                    "https://api.github.com/repos/civiform/civiform/branches/main",
+                    "commit", "sha")
+            elif "SNAPSHOT" in tag:
+                short_sha = tag.split("-")[1]
+                return self._fetch_json_val(
+                    f"https://api.github.com/repos/civiform/civiform/commits/{short_sha}",
+                    "sha")
+            else:
+                tag_url = self._fetch_json_val(
+                    f"https://api.github.com/repos/civiform/civiform/git/refs/tags/{tag}",
+                    "object", "url")
+                return self._fetch_json_val(tag_url, "object", "sha")
+        except self.VersionNotFoundError as e:
+            print(e)
+            return None
 
-    def _get_latest_version_number(self) -> str:
-        url = "https://api.github.com/repos/civiform/civiform/releases/latest"
+    def _fetch_json_val(self, url, field_one, field_two=None) -> str:
         response = requests.get(url)
         if response.status_code == 200:
-            return response.json()["tag_name"]
-        else:
-            raise self.VersionNotFoundError(
-                f"Error: 'latest' could not be translated to a release tag. {response.status_code} - {response.json()['message']}"
-            )
-
-    def _get_commit_sha_for_tag(self, tag_commit_sha: str) -> str:
-        tag_url = f"https://api.github.com/repos/civiform/civiform/git/tags/{tag_commit_sha}"
-        tag_response = requests.get(tag_url)
-        if tag_response.status_code == 200:
-            commit_sha = tag_response.json()["object"]["sha"]
-            return commit_sha
+            try:
+                return response.json(
+                )[field_one] if field_two is None else response.json(
+                )[field_one][field_two]
```

Comment:
**suggestion (blocking):** this is a little tricky to read, please rewrite to avoid having a conditional statement nested inside another statement
https://google.github.io/styleguide/pyguide.html#211-conditional-expressions

Reply:
done? lmk your thoughts.

Reply:
better, thanks!